app/src/main/java/com/sunasterisk/youtubebackground/screen/fragment/home/HomeFragment.kt

```diff
@@ -16,7 +26,34 @@ class HomeFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_home, container, false)
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initData()
+        initView()
+    }
+
+    private fun initData() {
+        val presenter = HomePresenter(VideoRepository.instance)
+        presenter.setView(this)
+        presenter.onStart()
+    }
+
+    private fun initView() {
+        recyclerHome.setHasFixedSize(true)
+        recyclerHome.adapter = adapter
+    }
+
     companion object {
         fun newInstance() = HomeFragment()
     }
+
+    override fun getPageSuccess(page: Page) {
+        page.videos?.let {
+            adapter.updateData(it)
+        }
+    }
+
+    override fun getPageFail(exception: Exception?) {
+        Toast.makeText(this.context, R.string.errorLoad, Toast.LENGTH_SHORT).show()
```

Comment:
context?.let {
    Toast.makeText(it, R.string.errorLoad, Toast.LENGTH_SHORT).show()
}